src/main/webapp/js/components/csv/csv-config-page.jsx

```diff
@@ -0,0 +1,259 @@
+import { getFieldConfigByEntity } from '../../selectors';
+import TextField from '../../utils/form/text-field';
+
+class CsvConfigPage extends Component {
+  onSubmit = (values) => {
+    const csvConfig = values;
+    csvConfig.entity = this.props.entityType;
+    this.props.onSubmit(csvConfig);
+  };
+
+  getOptions = () => _.map(this.props.fieldConfig, v => ({
+    label: v.displayName,
+    value: v.id,
+  }));
+
+  getFields = (fieldValue) => {
+    let hideFormat = true;
+
+    // eslint-disable-next-line max-len
+    const fieldConfig = this.props.fieldConfig[fieldValue.fieldConfigId];
+    if (fieldConfig && (fieldConfig.fieldType === DATE || fieldConfig.fieldType === DATE_TIME)) {
+      hideFormat = false;
+    }
+
+    return [
+      {
+        name: 'fieldConfigId',
+        fieldType: ENUM,
+        displayName: 'Field name',
+        required: true,
+        options: this.getOptions(),
+      },
+      {
+        name: 'fieldName',
+        fieldType: TEXT,
+        displayName: 'CSV field name',
+        required: true,
+      },
+      {
+        name: 'format',
+        fieldType: TEXT,
+        displayName: 'Format',
+        required: true,
+        hidden: hideFormat,
+      },
+      {
+        name: 'defaultValue',
+        fieldType: fieldConfig ? fieldConfig.fieldType : TEXT,
+        displayName: 'Default value',
+        relatedField: fieldConfig ? fieldConfig.relatedField : null,
+        relatedEntity: fieldConfig ? fieldConfig.relatedEntity : null,
+        required: fieldConfig ? fieldConfig.required : false,
+        format: fieldConfig ? fieldConfig.format : null,
+      },
+    ];
+  };
+
+  getRelatedFields = (relatedField, relatedEntity) => [
+    {
+      name: 'relatedField',
+      fieldType: RELATION,
+      displayName: 'Related field',
+      required: true,
+      relatedField,
+      relatedEntity,
+    },
+    {
+      name: 'fieldValue',
+      fieldType: TEXT,
+      displayName: 'Related field CSV name',
+      required: true,
+    },
+  ];
+
+  validate = (values) => {
+    const errors = {};
+    errors.csvFields = [];
+
+    _.forEach(values.csvFields, (csvField, key) => {
+      _.forEach(this.getFields(values.csvFields), (config) => {
+        const val = csvField[config.name];
+
+        if (config.required && !config.hidden && (_.isNil(val) || val === '')) {
+          errors.csvFields[key] = { [config.name]: `${config.displayName} is required` };
+        }
+      });
+    });
+
+    return errors;
+  };
+
+  render() {
+    const { entityType } = this.props;
+
+    return (
+      <div>
+        <Form
+          onSubmit={this.onSubmit}
+          initialValues={this.props.csvConfig}
+          validate={this.validate}
+          mutators={{
+            ...arrayMutators,
+          }}
+          render={({
+            handleSubmit, values, invalid,
+          }) => (
+            <form onSubmit={handleSubmit}>
+              { !_.isEmpty(this.props.csvConfig) ? (
```

Comment:
I think you can add a new property for this 'isNew' or something like this and set it to true in CsvConfigCreate, also you can just disable the TextField instead of rendering separate components, the code will look much cleaner